### ISSUE TITLE
getBoundingClientRect() always fails the first time with 'undefined' error

### ISSUE THREAD

**Issue by Avinodh** [nhn/tui.calendar#1357]
## Version
2.1.3

## Test Environment
Google Chrome

## Current Behavior
I am trying to configure custom popups for event creation. The following code sets up the event handler to listen on the `selectDateTime` event:
```
        this.calendarInstance.on('selectDateTime', (info) => {
            console.log(info.start + " - " + info.end);
            var gridElems = info.gridSelectionElements;
            console.log(gridElems[0].getBoundingClientRect());
        });
```

The first time I  click the calendar, I  get the following error:
<img width="1561" alt="image" src="https://user-images.githubusercontent.com/5704154/213888749-9c2e6919-c464-4444-976e-1378c4ae4c96.png">

However, any subsequent clicks seem to work. just fine (the below screenshot shows that the second click did in fact print the boundingRect DOM element):
<img width="1106" alt="image" src="https://user-images.githubusercontent.com/5704154/213888765-0a64e2e0-7b92-4a93-9963-d2b1a1b708f9.png">

My guess is that there is some sort of a race here between when the first event gets fired off and the DOM being ready. But why does this consistently happen on only the first click?

**Comment by adhrinae** [comment]
@Avinodh 

You are correct, it is a bug caused by a race condition that occurs when selecting datetime by clicking on cells.

Unlike dragging, the calendar needs to determine whether it is a single or double click. Once the calendar recognizes the click interaction, it draws the selection element and re-renders. [(code)](https://github.com/nhn/tui.calendar/blob/main/apps/calendar/src/hooks/gridSelection/useGridSelection.ts#L121-L123)

However, the gridSelectionElements refer to variables declared during the previous render phase, which is why the proper gridSelectionElements property cannot be obtained.

This may seem peculiar, but it is how React (the calendar uses Preact internally) works.

Resolving this issue would be pretty difficult. As a workaround, you can disable clicks through an option and only allow users to drag cells.

**Comment by Avinodh** [comment]
Thanks @adhrinae ! I'll look into disabling the click event, and just going with the drag behavior.

**Comment by adhrinae** [comment]
@Avinodh I'll leave this issue open since it's not resolved yet.
